org.lflang/src/org/lflang/generator/ActionInstance.java

```diff
@@ -0,0 +1,123 @@
+ */
+public class ActionInstance extends TriggerInstance<Action> {
+    
+    /**
+     * Create a new timer instance.
+     * If the definition is null, then this is a shutdown action.
+     * @param definition The AST definition, or null for startup.
+     * @param parent The parent reactor.
+     */
+    public ActionInstance(Action definition, ReactorInstance parent) {
+        super(definition, parent);
+        if (parent == null) {
+            throw new InvalidSourceException("Cannot create an ActionInstance with no parent.");
+        }
+        if (definition != null) {
+            if (definition.getMinDelay() != null) {
+                Parameter parm = definition.getMinDelay().getParameter();
+                if (parm != null) {
+                    this.minDelay = ASTUtils.getTimeValue(
+                            parent.lookupParameterInstance(parm).init.get(0));
+                } else {
+                    this.minDelay = ASTUtils.getTimeValue(definition.getMinDelay());
+                }
+            }
+            if (definition.getMinSpacing() != null) {
+                Parameter parm = definition.getMinSpacing().getParameter();
+                if (parm != null) {
+                    this.minSpacing = ASTUtils.getTimeValue(
+                            parent.lookupParameterInstance(parm).init.get(0));
+                } else {
+                    this.minSpacing = ASTUtils.getTimeValue(definition.getMinSpacing());
+                }
+            }
+            if (definition.getOrigin() == ActionOrigin.PHYSICAL) {
+                physical = true;
+            }
+            policy = definition.getPolicy();
+        }
+    }
+    
+    //////////////////////////////////////////////////////
+    //// Public fields.
+
+    /** The constant default for a minimum delay. */
+    public static TimeValue DEFAULT_MIN_DELAY = new TimeValue(0, TimeUnit.NONE);
+        
+    //////////////////////////////////////////////////////
+    //// Public methods.
+
+    /** Return the minimum delay for this action. */
+    public TimeValue getMinDelay() {
+        return minDelay;
+    }
+    
+    /** Return the minimum spacing between action events. */
+    public TimeValue getMinSpacing() {
+        return minSpacing;
+    }
+    
+    /** Return the replacement policy for when minimum spacing is violated. */
+    public String getPolicy() {
+        return policy;
+    }
+    
+    /** Return the indicator of whether the action is physical. */
+    public boolean isPhysical() {
+        return physical;
+    }
+
+    //////////////////////////////////////////////////////
+    //// Protected fields.
+
+    /** The minimum delay for this action. */
+    TimeValue minDelay = DEFAULT_MIN_DELAY;
+    
+    /** The minimum spacing between action events. */
+    TimeValue minSpacing = null;
+    
+    /** The replacement policy for when minimum spacing is violated. */
+    String policy = null;
```

Comment:
Should the default be null?

Reply:
This is set unconditionally in the constructor, so null is not the default.